src/login/logind-session.c

```diff
@@ -1096,6 +1113,7 @@ static int session_dispatch_fifo(sd_event_source *es, int fd, uint32_t revents,
         /* EOF on the FIFO means the session died abnormally. */
 
         session_remove_fifo(s);
+        session_invalidate_leader(s);
```

Comment:
Shouldn't this be called from `session_stop()` instead?

Reply:
It will be called eventually after scope stop job actually finished and then leader process should be gone. Calling it from `session_stop()` seem a bit early to me.